test/fail_compilation/fail18057.d

```diff
@@ -0,0 +1,16 @@
+/**
+TEST_OUTPUT:
+---
+fail_compilation/fail18057.d(13): Error: variable fail18057.RBNode!int.RBNode.copy recursive initialization of field
+fail_compilation/fail18057.d(16): Error: template instance fail18057.RBNode!int error instantiating
+---
+*/
+
+// https://issues.dlang.org/show_bug.cgi?id=18057
+// Recursive field initializer causes segfault.
+struct RBNode(T)
```

Comment:
Doesn't need to be a template.  Only did that for the benefit of `__traits(compiles)`.

Reply:
OK. Removed.